src/zope/tales/tests/test_expressions.py

```diff
@@ -261,15 +316,49 @@ def check(expr):
 
     def test_defer_expression_returns_wrapper(self):
         from zope.tales.expressions import DeferWrapper
+        from zope.tales.expressions import DeferExpr
+        expr = self.engine.compile('defer: b')
+        self.assertIsInstance(expr, DeferExpr)
+        self.assertEqual(str(expr), "<DeferExpr 'b'>")
+        self._check_evals_to_instance(expr, DeferWrapper)
+
+        wrapper = expr(self.context)
+        self.assertEqual(wrapper(), self.context.vars['b'])
+
+        self.assertEqual(str(wrapper), 'boot')
+
+    def test_eval_defer_wrapper(self):
         expr = self.engine.compile('defer: b')
-        context=self.context
-        self.assertTrue(isinstance(expr(context), DeferWrapper))
+        self.context.vars['deferred'] = expr(self.context)
+        self._check_evals_to('deferred', self.context.vars['b'])
 
     def test_lazy_expression_returns_wrapper(self):
         from zope.tales.expressions import LazyWrapper
+        from zope.tales.expressions import LazyExpr
         expr = self.engine.compile('lazy: b')
-        context=self.context
-        self.assertTrue(isinstance(expr(context), LazyWrapper))
+        self.assertIsInstance(expr, LazyExpr)
+        self.assertEqual(repr(expr), "lazy:'b'")
+        lazy = expr(self.context)
+        self.assertIsInstance(lazy, LazyWrapper)
+
+        first_result = lazy()
+        second_result = lazy()
+        self.assertIs(first_result, second_result)
+
+
+    def test_not(self):
+        def evaluateBoolean(expr):
+            return bool(expr(self.context))
+        self.context.evaluateBoolean = evaluateBoolean
+        self._check_evals_to('not:exists:x', 0)
+        expr = self._check_evals_to('not:exists:v_42', 1)
```

Comment:
Does `not:` return actual ints, or does it return bools that compare equal to 0/1?

Reply:
It [also returns ints](https://github.com/zopefoundation/zope.tales/blob/master/src/zope/tales/expressions.py#L286).